src/components/data-table-v2/_data-table-v2-core.scss

```diff
@@ -185,13 +185,13 @@
   //----------------------------------------------------------------------------
   // Table title text
   //----------------------------------------------------------------------------
-  .#{$prefix}--data-table-v2-header {
+  .#{$prefix}--data-table-v2-title-container {
```

Comment:
If it helps, this made a lot of sense to me as `header` and then `title` and maybe `description` instead of optional.

I don't think changing the class name would be too bad, and semantically the top part is a header for the section 👍